File: src/app/Model/Message.ts
export class Message {
  private idMessage:number;
  private senderId: string;
  private content: string;
  private sendDate: Date;
  private contentType: string;
  private messageType : string;

  constructor(idMessage:number,sendDate: Date, messageType: string,  contentType: string, content: string,senderId: string) {
    this.senderId = senderId;
    this.content = content;
    this.sendDate = sendDate;
    this.contentType = contentType;
    this.messageType = messageType;
    this.idMessage = idMessage;
  }

  // Getter and Setter for 'senderId'
  get SenderId(): string {
    return this.senderId;
  }

  set SenderId(value: string) {
    this.senderId = value;
  }

  // Getter and Setter for 'content'
  get Content(): string {
    return this.content;
  }

  set Content(value: string) {
    this.content = value;
  }

  // Getter and Setter for 'state'


  // Getter and Setter for 'sendDate'
  get SendDate(): Date {
    return this.sendDate;
  }

  set SendDate(value: Date) {
    this.sendDate = value;
  }
}

File: src/main/java/com/example/messenger/POJO/Message.java
package com.example.messenger.POJO;

import com.example.messenger.Config.MessageUserSerializer;
import com.example.messenger.POJO.Enums.MessageType;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import jakarta.persistence.*;
import lombok.Data;

import java.sql.Timestamp;

@Entity
@Data

public class Message {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long idMessage;

    @ManyToOne
    @JoinColumn(name = "sender_id")
    @JsonSerialize(using = MessageUserSerializer.class)
    @JsonProperty("senderId")
    User sender;
    @ManyToOne
    @JsonSerialize(using = MessageUserSerializer.class)
    @JsonProperty("receiverId")
    @JoinColumn(name = "receiver_id")
    User receiver;


    MessageType type;
    private Timestamp sendDate;
    private String contentType;
    public Message() {
        // Default constructor
    }

    public Message( MessageType type, Timestamp sendDate) {

        this.type = type;
        this.sendDate = sendDate;

    }
    public Message(User sender,User receiver, MessageType type, Timestamp sendDate) {
        this.sender=sender;
        this.receiver=receiver;
        this.type = type;
        this.sendDate = sendDate;
    }
    public Message(User sender,User receiver, MessageType type, Timestamp sendDate,String contentType) {
        this.sender=sender;
        this.receiver=receiver;
        this.type = type;
        this.sendDate = sendDate;
        this.contentType=contentType;
    }

}
